src/main/java/slimeknights/tconstruct/library/recipe/tinkerstation/building/ToolBuildingRecipe.java

```diff
@@ -38,6 +39,7 @@ public class ToolBuildingRecipe implements ITinkerStationRecipe {
     LoadableRecipeSerializer.RECIPE_GROUP,
     TinkerLoadables.MODIFIABLE_ITEM.requiredField("result", r -> r.output),
     IntLoadable.FROM_ONE.defaultField("result_count", 1, true, r -> r.outputCount),
+    ResourceLocationLoadable.DEFAULT.defaultField("slot_id", r -> r.getId, r -> r.slotId),
```

Comment:
This is not syntactically valid, a defaulting field has to default to a value, not a function. Just use a nullable field. 

You can either map null to the value in the constructor (remove `@AllArgsConstructor` and use the default constructor).

Since you are adding a new constructor, keep the old signature (without the layout ID) around but marked as deprecated calling the new one and redirect to the new